### PR TITLE
Issues/75 folder contents data

### PR DESCRIPTION
**Pull request by aerych**
Refs #75 

This PR is the first part of adding a core data proxy layer to story assets.  In this PR we:
- Roll back earlier temporary code that interacted directly with the file system
- Implement an AssetStore to wrangle the StoryAsset model
- Add the ability to create a rudimentary text note. This will be expanded on in its own focused sprint.
- Wire up basic display, create, and delete functionality for the AssetViewController

Future PRs will introduce sorting and ordering of the list, and core data/file system reconciliation. 

To test:
- View the list of story folders.
- Add some extra folders if necessary so you have more than one.
- Tap into a folder that is not the currently selected one (this should automatically select it).  You should now be viewing the Assets controller for the folder you selected.
- Use the plus button to add a few assets. Each asset should be shown as a row titled "New Text Note".
- Swipe to delete one of the rows. Confirm the delete works. 

@jleandroperez Can I trouble you with a review of this one?

### CODE REVIEW

File: Newspack/Newspack/Controllers/AssetsViewController.swift [hunk 1]
```diff
@@ -1,39 +1,129 @@
...
 
-    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "AssetCell", for: indexPath)
+// MARK: - DataSource related methods
+extension AssetsViewController {
 
-        let url = items[indexPath.row]
-        cell.textLabel?.text = url.lastPathComponent
+    func cellFor(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "AssetCell", for: indexPath)
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/AssetsViewController.swift:40]: **No need to change anything!!!** I love [this trick](https://github.com/Automattic/simplenote-macos/blob/develop/Simplenote/NSTableViewCell+Simplenote.swift#L11)!!!

File: Newspack/Newspack/Controllers/AssetsViewController.swift [hunk 2]
```diff
@@ -1,39 +1,129 @@
...
+
+// MARK: - AssetDataSource
+class AssetDataSource: UITableViewDiffableDataSource<AssetDataSource.Section, StoryAsset> {
+
+    enum Section: CaseIterable {
+        case main
+    }
+
+    // Receipt so we can respond to any emitted changes in the AssetStore.
+    var receipt: Any?
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/AssetsViewController.swift]: Maybe `private` ?

**Review by jleandroperez — APPROVED** [review]
✅  Verified: AssetsViewController is pushed
✅  Verified: New assets are added after pressing `+`
✅  Verified: Assets can be swiped / deleted
✅  Verified: Unit tests are green
✅  Verified: No regressions in folders management

Looking sharp @aerych!!!

:shipit:

File: Newspack/Newspack/Controllers/AssetsViewController.swift (hunk 1 above)
Comment (aerych) [reply Newspack/Newspack/Controllers/AssetsViewController.swift:40]: I [like that trick too](https://github.com/Automattic/newspack-ios/blob/develop/Newspack/Newspack/Extensions/UITableViewCell%2BHelpers.swift). :D  For now this is just a regular UITableViewCell vs a custom class. When I create the custom cells I'll up date this too.   Also props to you for originally suggesting the trick several PRs ago!

File: Newspack/Newspack/Controllers/AssetsViewController.swift (hunk 2 above)
Comment (aerych) [reply Newspack/Newspack/Controllers/AssetsViewController.swift]: Private indeed

**Pull request merged**